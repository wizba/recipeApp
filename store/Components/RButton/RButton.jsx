import React, { useEffect } from "react";
import { Pressable, StyleSheet, Text, View } from "react-native";

function RButton({ externalStyles, text, onPress }) {
  return (
    <View style={[externalStyles?.container, styles.container]}>
      <Pressable
        style={[externalStyles?.button, styles.button]}
        onPress={onPress}
        android_ripple={{ color: '#dddddd' }}
        testID="RButton"
      >
        <Text style={[externalStyles?.text, styles.text]}>{text}</Text>
        <Text style={[externalStyles?.text, styles.text]}>{text}</Text>
      </Pressable>
    </View>
  );
}

export default RButton;

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#9b9898",
    width: "100%",
    borderRadius: 10,
    overflow: "hidden",
     elevation: 4,
  },
  button: {
    width: "100%",
    borderRadius: 10,
    overflow: "hidden",
    padding: 10,
    justifyContent: "center",
    alignItems: "center",
  },
  text: {
    fontSize: 32,
    color: "#1d1d1d",
  },
});
