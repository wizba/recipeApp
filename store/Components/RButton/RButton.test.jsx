import { render, screen, fireEvent } from "@testing-library/react-native";
import RButton from "./RButton";

describe("RButton Component", () => {
  test("renders correctly", () => {
    const tree = render(<RButton />);
    expect(tree).toMatchSnapshot();
  });

  test("renders the correct text", () => {
    render(<RButton text="Test" />);
    expect(screen.getByText("Test")).toBeTruthy();
  });

  test('renders the correct onPress', () => {
    const onPressMock = jest.fn();
    render(<RButton onPress={onPressMock} text="Click Me" />);
  
    fireEvent.press(screen.getByTestId('RButton'));
  
    expect(onPressMock).toHaveBeenCalled();
  });
});

