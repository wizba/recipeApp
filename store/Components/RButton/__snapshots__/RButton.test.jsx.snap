// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RButton Component renders correctly 1`] = `
<View
  style={
    [
      undefined,
      {
        "backgroundColor": "#9b9898",
        "borderRadius": 10,
        "elevation": 4,
        "overflow": "hidden",
        "width": "100%",
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        undefined,
        {
          "alignItems": "center",
          "borderRadius": 10,
          "justifyContent": "center",
          "overflow": "hidden",
          "padding": 10,
          "width": "100%",
        },
      ]
    }
    testID="RButton"
  >
    <Text
      style={
        [
          undefined,
          {
            "color": "#1d1d1d",
            "fontSize": 32,
          },
        ]
      }
    />
  </View>
</View>
`;
